### PR TITLE
[libra framework] Specified the access control properties of the txn scripts

### PR DESCRIPTION
**Pull request by junkil-park**
- added the access control specs for the P1/P2/P3 transaction scripts

## Motivation

To specify the access control specs of the transaction scripts

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

cargo test

### CODE REVIEW

File: language/stdlib/transaction_scripts/rotate_authentication_key_with_nonce_admin.move [hunk 1]
```diff
@@ -60,6 +61,8 @@ spec fun rotate_authentication_key_with_nonce_admin {
         Errors::NOT_PUBLISHED; // TOOD: Undocumented error code. Added due to the possible absence of SlidingNonce in SlidingNonce::try_record_nonce;
 
     /// Access Control
+    /// Only the Libra Root account can process the admin scripts [[H8]][PERMISSION].
+    requires Roles::has_libra_root_role(lr_account); /// This is ensured by LibraAccount::writeset_prologue.
```
Comment (sblackshear) [inline language/stdlib/transaction_scripts/rotate_authentication_key_with_nonce_admin.move:65]: Would it make more sense to require the entire `spec fun` for `writeset_prologue` here? That would make the comment unnecessary + let the script assume other properties that are true and may be needed in the future.

File: language/stdlib/transaction_scripts/set_validator_operator_with_nonce_admin.move [hunk 2]
```diff
@@ -80,5 +82,11 @@ spec fun set_validator_operator_with_nonce_admin {
         Errors::INVALID_ARGUMENT,
         Errors::NOT_PUBLISHED,
         Errors::REQUIRES_ROLE;
-    }
+
+    /// Access Control
+    /// Only the Libra Root account can process the admin scripts [[H8]][PERMISSION].
+    requires Roles::has_libra_root_role(lr_account); /// This is ensured by LibraAccount::writeset_prologue.
```
Comment (sblackshear) [inline language/stdlib/transaction_scripts/set_validator_operator_with_nonce_admin.move:88]: Same question re: prologue here.

**Review by sblackshear — APPROVED** [review]
This is excellent, thank you!!

File: language/stdlib/transaction_scripts/rotate_authentication_key_with_nonce_admin.move (hunk 1 above)
Comment (junkil-park) [reply language/stdlib/transaction_scripts/rotate_authentication_key_with_nonce_admin.move:65]: That's a good point. Thanks to @emmazzz , each txn script already has the following line in order to "require" other properties that the prologue function ensures:
```
include LibraAccount::TransactionChecks{sender: account}; // properties checked by the prologue.
```

File: language/stdlib/transaction_scripts/set_validator_operator_with_nonce_admin.move (hunk 2 above)
Comment (junkil-park) [reply language/stdlib/transaction_scripts/set_validator_operator_with_nonce_admin.move:88]: Same as above!

File: language/stdlib/transaction_scripts/rotate_authentication_key_with_nonce_admin.move (hunk 1 above)
Comment (sblackshear) [reply language/stdlib/transaction_scripts/rotate_authentication_key_with_nonce_admin.move:65]: The problem is that admin scripts *do not* run the ordinary transaction prologue, so it is not correct to assume this. Instead, admin scripts should assume that the writeset prologue is run.

**Pull request closed without merging**